### PR TITLE
Only changes power profile, if not current.

### PR DESCRIPTION
**Pull request by jsj1027**
Queries the current power profile before attempting to change it.
If it is the requested change is the same as the current profile, the system will not attempt to switch.

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -38,18 +38,36 @@ impl PowerClient {
...
-        println!("setting power profile to battery");
-        self.call_method::<bool>("Battery", None).map(|_| ())
+        let profile = self.get_profile().ok();
+        let profile = profile.as_ref().map_or("?", |s| s.as_str());
+        if profile != "Battery" {
+            println!("setting power profile to battery");
+            self.call_method::<bool>("Battery", None).map(|_| ())
+        } else {
+            Err(String::from("power profile is already set to battery"))
+        }
```
Comment (mmstick) [inline src/client.rs]: We could condense these down to:

```rust
fn balanced(&mut self) -> Result<(), String> {
    self.profile("Balanced")
}

....

fn profile(&mut self, profile: &str) -> Result<(), String> {
    let current = self.get_profile().ok();
    let current = profile.as_ref().map_or("?", |s| s.as_str());
    if current != profile {
        println!("setting power profile to {}", profile);
        self.call_method::<bool>(profile, None).map(|_| ())
    else {
        Err(format!("power profile is already set to {}", profile))
    }
}
```

Although we may not want to return an error for a profile that's already set. Returning `Ok(())` should be fine.

Comment (jsj1027) [reply src/client.rs]: Sounds good ill make the changes

**Comment by jsj1027** [comment]
:D

**Comment by jsj1027** [comment]
Is there anything else I need to do to help this commit along? Or is everything okay?

**Comment by brs17** [comment]
> Is there anything else I need to do to help this pull request along? Or is everything okay?

Everything should be good from your side, currently QA has other things higher up in the queue before getting to this PR. Thank you for your patience!

**Pull request merged**